buildings_bench/models/resnet.py

```diff
@@ -0,0 +1,137 @@
+import torch
+from buildings_bench.models.base_model import BaseSurrogateModel
+from buildings_bench.models.transformers import TimeSeriesSinusoidalPeriodicEmbedding
+import torch.nn as nn
+import torch.nn.functional as F
+import numpy as np
+from typing import Tuple, Dict
+
+'''
+Basic Residual Block
+'''
+class ResBlock(nn.Module):
+
+    def __init__(self, in_dim, out_dim, downsample=None):
+        super(ResBlock, self).__init__()
+        self.activation = nn.ReLU()
+        self.linear1 = nn.Linear(in_dim, out_dim)
+        self.linear2 = nn.Linear(out_dim, out_dim)
+        self.downsample = downsample
+
+    def forward(self, x):
+        identity = x
+        
+        out = self.linear1(x)
+        out = self.activation(out)
+        out = self.linear2(out)
+        
+        if self.downsample is not None:
+            identity = self.downsample(x)
+        
+        out += identity
+        out = self.activation(out)
+
+        return out
+
+'''
+ResNet 
+'''
+class ResNet(BaseSurrogateModel):
+
+    def __init__(self, hidden_dims,
+                pred_len=1,
+                continuous_head='mse',
+                continuous_loads=True):
+        
+        super(ResNet,self).__init__(pred_len, continuous_loads)
+        self.continuous_head = continuous_head
+        out_dim = 1 if self.continuous_head == 'mse' else 2
+        self.logits = nn.Linear(hidden_dims[-1], out_dim)
+        
+        self.building_embedding = nn.Embedding(2, 32)
```

Comment:
I think moving forward let's remove these features-- "building_embedding" (which is binary 0 or 1 for res/com), "lat_embedding" , and "lon_embedding". These infos are going to be in the building characteristic feature vector /text caption now, unlike in BuildingsBench where we don't have this

Reply:
Removed!